internal/buildinfo/version.go

```diff
@@ -2,23 +2,164 @@ package buildinfo
 
 import (
 	"errors"
-	"fmt"
-	"os"
 	"runtime/debug"
-	"strconv"
+	"strings"
 	"time"
 
 	"github.com/blang/semver"
+	"github.com/loadsmart/calver-go/calver"
 )
 
+type Version interface {
+	String() string
+	EQ(other Version) bool
+	Newer() bool
+	SeverelyOutdated(other Version) bool
+}
+
+type SemverVersion struct {
+	semver.Version
+}
+
+func ParseSemver(version string) (*SemverVersion, error) {
+	v, err := semver.ParseTolerant(version)
+	if err != nil {
+		return nil, err
+	}
+	return &SemverVersion{v}, nil
+}
+
+func (v SemverVersion) EQ(other Version) bool {
+	o, ok := other.(*SemverVersion)
+	if ok {
+		return v.Version.EQ(o.Version)
+	} else {
+		return false
+	}
+}
+
+func (v *SemverVersion) Newer() bool {
+	_, ok := parsedVersion.(*CalverVersion)
+	if ok {
+		return false
+	} else {
+		other := parsedVersion.(*SemverVersion)
+		return v.Version.GT(other.Version)
+	}
+}
+
+func (v *SemverVersion) SeverelyOutdated(other Version) bool {
+	otherVer, ok := other.(*SemverVersion)
+	if ok {
+		if v.Major < otherVer.Major {
+			return true
+		}
+		if v.Minor < otherVer.Minor {
+			return true
+		}
+		if v.Patch+5 < otherVer.Patch {
+			return true
+		}
+		return false
+	} else {
+		return false
```

Comment:
If we return true when V is older than otherVer, shouldn't we return true if other is not semver?

That would imply V is semver but other is calver

Reply:
Almost certainly, my brain was coffee-deprived when I wrote that. I'll check it over in a bit.